bundles/org.openhab.binding.modbus.sbc/src/main/java/org/openhab/binding/modbus/sbc/internal/ALD1Registers.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.sbc.internal;
+
+import static org.openhab.io.transport.modbus.ModbusConstants.ValueType.*;
+
+import java.math.BigDecimal;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.library.unit.SmartHomeUnits;
+import org.openhab.io.transport.modbus.ModbusConstants;
+import org.openhab.io.transport.modbus.ModbusConstants.ValueType;
+
+/**
+ * The {@link ALD1Registers} is responsible for defining Modbus registers and their units.
+ *
+ * @author Fabian Wolter - Initial contribution
+ */
+@NonNullByDefault
+public enum ALD1Registers {
+    TOTAL_ENERGY(0.01f, 28, UINT32, SmartHomeUnits.KILOWATT_HOUR),
+    PARTIAL_ENERGY(0.01f, 30, UINT32, SmartHomeUnits.KILOWATT_HOUR), // only unidirectional meters
+    FEEDING_BACK_ENERGY(0.01f, 30, UINT32, SmartHomeUnits.KILOWATT_HOUR), // only bidirectional meters
+    VOLTAGE(1, 36, UINT16, SmartHomeUnits.VOLT),
+    CURRENT(0.1f, 37, UINT16, SmartHomeUnits.AMPERE),
+    ACTIVE_POWER(10, 38, INT16, SmartHomeUnits.WATT),
+    REACTIVE_POWER(10, 39, INT16, SmartHomeUnits.VAR),
+    POWER_FACTOR(0.01f, 40, UINT16, SmartHomeUnits.ONE);
+
+    private BigDecimal multiplier;
+    private int registerAddress;
+    private ModbusConstants.ValueType type;
+    private Unit<?> unit;
+
+    private ALD1Registers(float multiplier, int registerAddress, ValueType type, Unit<?> unit) {
+        this.multiplier = new BigDecimal(multiplier);
```

Comment:
```suggestion
    TOTAL_ENERGY("0.01", 28, UINT32, SmartHomeUnits.KILOWATT_HOUR),
    PARTIAL_ENERGY("0.01", 30, UINT32, SmartHomeUnits.KILOWATT_HOUR), // only unidirectional meters
    FEEDING_BACK_ENERGY("0.01", 30, UINT32, SmartHomeUnits.KILOWATT_HOUR), // only bidirectional meters
    VOLTAGE("1", 36, UINT16, SmartHomeUnits.VOLT),
    CURRENT("0.1", 37, UINT16, SmartHomeUnits.AMPERE),
    ACTIVE_POWER("10", 38, INT16, SmartHomeUnits.WATT),
    REACTIVE_POWER("10", 39, INT16, SmartHomeUnits.VAR),
    POWER_FACTOR("0.01", 40, UINT16, SmartHomeUnits.ONE);

    private BigDecimal multiplier;
    private int registerAddress;
    private ModbusConstants.ValueType type;
    private Unit<?> unit;

    private ALD1Registers(String multiplier, int registerAddress, ValueType type, Unit<?> unit) {
        this.multiplier = new BigDecimal(multiplier);
```

Just a cosmetic suggestion 

With this change you would have "neat" finite numbers since big decimal operate in 10-base. See https://stackoverflow.com/questions/7408566/java-double-value-0-01-changes-to-0-009999999999999787 

Feel free to ignore this comment :)

Reply:
I understand, but it seems more odd to me to have these numeric constants as strings...

Reply:
Yup.. No way around the strings really if we want stay fully in BigDecimal domain. 

As said, feel free to ignore this comment, makes little practical difference really